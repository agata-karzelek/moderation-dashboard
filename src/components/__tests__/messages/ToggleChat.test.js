import React from "react";
import { shallow } from "enzyme";
import ToggleChat from "../../messages/ToggleChat";

const mockToggleFn = jest.fn();
describe("Test Case For Toggle Chat", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<ToggleChat toggledVal="chat" setToggledVal={mockToggleFn} />);
  });

  test("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("initial styling", () => {
    expect(wrapper.find("#chat").getElement().props.className.includes("activeToggled")).toBe(true);

    expect(wrapper.find("#ban").getElement().props.className.includes("disableToggled")).toBe(true);
  });

  test("check function", () => {
    expect(mockToggleFn).toHaveBeenCalledTimes(0);

    wrapper.find("#chat").simulate("click");
    expect(mockToggleFn).toHaveBeenCalledTimes(1);

    wrapper.find("#ban").simulate("click");
    expect(mockToggleFn).toHaveBeenCalledTimes(2);
  });
});
