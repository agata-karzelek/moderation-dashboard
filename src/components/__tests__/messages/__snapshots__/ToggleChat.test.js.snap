// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Case For Toggle Chat Snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-activeToggled-23"
    id="chat"
    onClick={[Function]}
  >
    Chat
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-disableToggled-24"
    id="ban"
    onClick={[Function]}
  >
    Banned
  </WithStyles(ForwardRef(Button))>
</Fragment>
`;
